src/test/java/com/github/checkstyle/regression/extract/ExtractInfoProcessorTest.java

```diff
@@ -0,0 +1,88 @@
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.regression.extract;
+
+import static com.github.checkstyle.regression.internal.TestUtils.assertUtilsClassHasPrivateConstructor;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.spy;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.Map;
+
+import org.junit.Test;
+
+import com.github.checkstyle.regression.data.ImmutableModuleExtractInfo;
+import com.github.checkstyle.regression.data.ModuleExtractInfo;
+
+public class ExtractInfoProcessorTest {
+    private static final String BASE_PACKAGE = "com.puppycrawl.tools.checkstyle";
+
+    @Test
+    public void testIsProperUtilsClass() throws Exception {
+        assertUtilsClassHasPrivateConstructor(ExtractInfoProcessor.class);
+    }
+
+    @Test
+    public void testGetNameToModuleInfoFromReader() {
+        final InputStream is = ExtractInfoProcessor.class.getClassLoader()
+                .getResourceAsStream("checkstyle_modules.json");
+        final InputStreamReader reader = new InputStreamReader(is, Charset.forName("UTF-8"));
+        final Map<String, ModuleExtractInfo> map =
+                ExtractInfoProcessor.getNameToModuleExtractInfoFromReader(reader);
+
+        final String module1 = "NewlineAtEndOfFileCheck";
+        final ModuleExtractInfo extractInfo1 = ImmutableModuleExtractInfo.builder()
+                .name(module1)
+                .packageName(BASE_PACKAGE + ".checks")
+                .parent("Checker")
+                .build();
+        final String module2 = "EmptyStatementCheck";
+        final ModuleExtractInfo extractInfo2 = ImmutableModuleExtractInfo.builder()
+                .name(module2)
+                .packageName(BASE_PACKAGE + ".checks.coding")
+                .parent("TreeWalker")
+                .build();
+        assertEquals(2, map.size());
```

Comment:
We started requiring assertion messages in main repo by PMD.
This project doesn't have the newly updated PMD, but I may add it soon.

Please make sure all new assertions in PRs have message. It can be very simple phrase.
Check out main project for any examples.